feat(protocol-designer): Redesign modals for temperature pauses

SyntaxColoring opened the pull request:
## Overview

Closes EXEC-1907.

## Changelog

Behind the concurrent module actions feature flag:

* Update the copy of the existing modals that show up after you save a Temperature Module or Heater-Shaker step.
* Remove the ability to skip adding the pause step.
* Add new modals for Thermocycler lid, block, and profile steps. These modals aren't used anywhere yet.

## Test Plan and Hands on Testing

* [x] The new modals match designs.
* [x] The new modals always add pause steps.
* [x] The new modals show up when the concurrent module actions feature flag is enabled, and not when it's disabled.

<img width="509" height="258" alt="Screenshot 2025-09-30 at 11 59 07 AM" src="https://github.com/user-attachments/assets/081b0fe0-0a47-4460-aad2-399ec64bd815" />
<img width="506" height="259" alt="Screenshot 2025-09-30 at 11 58 22 AM" src="https://github.com/user-attachments/assets/0cd3f005-ebb1-441f-b0a9-e54d6b5e57cc" />


## Review requests

None in particular.

## Risk assessment

Low.

SyntaxColoring committed:
Delete obsolete unused localization entries.

SyntaxColoring committed:
Add spaces before "°" symbols.

SyntaxColoring committed:
Rename "redesign" -> "legacy".

SyntaxColoring committed:
Type displayModule as required.

SyntaxColoring committed:
Add new copy.

SyntaxColoring committed:
Flesh out component.

SyntaxColoring committed:
Rename cancel/continue to skip-pause/add-pause.

SyntaxColoring committed:
Move doc comment so it actually shows up on the thing it's documenting.

SyntaxColoring committed:
Use non-breaking spaces for temperatures.

SyntaxColoring committed:
Fix blank temperature in legacy modal.

SyntaxColoring commented on protocol-designer/src/assets/localization/en/modal.json:
I'm deliberately "hard-coding" the module name in all of these localization strings, instead of interpolating it as `{{module}}`, to try to follow https://www.i18next.com/principles/best-practices.

SyntaxColoring commented on components/src/hooks/useConditionalConfirm.ts:
Just moving the doc comment in this file so it shows up when you hover over `useConditionalConfirm` in your IDE.

SyntaxColoring committed:
Consistent treatment of nullability.

koji commented on protocol-designer/src/components/organisms/AutoAddPauseUntilTempStepModal/index.tsx:
```diff
@@ -1,74 +1,199 @@
-  } = props
-  const { t } = useTranslation('modal')
+  const { type, handleSkipPauseClick, handleAddPauseClick } = props
+  const { t } = useTranslation()
```
not pointing out any file. is this intentional?

SyntaxColoring replied:
It is intentional, but I don't know if it's a good idea. I did it because I'm using a mixture of `modal.json`, `alert.json`, and `common.json`. So it felt clearer to me to explicitly specify the file everywhere (like `"modal:auto_add_pause_until_temp_step.legacy.title"`) instead of up top here.

Is there a downside to not listing the files in `useTranslation()`?

jerader replied:
wow, i thought you needed to list an array of each file you are using haha. if it works without adding the list, then i guess this is fine! just doesn't follow our pattern 😄

SyntaxColoring replied:
Yeah, I'll be honest, I really don't understand the implications of passing a list to `useTranslation()`. I'll bring it up in Slack.

koji commented on protocol-designer/src/components/organisms/AutoAddPauseUntilTempStepModal/index.tsx:
```diff
@@ -1,74 +1,199 @@
 
+type AutoAddPauseUntilTempStepModalProps =
+  | ({
+      type:
```
recommend using ◯◯Type instead `type`

SyntaxColoring replied:
Rename from `type` to `modalType`, let me know if that's better.

koji commented on protocol-designer/src/components/organisms/AutoAddPauseUntilTempStepModal/index.tsx:
```diff
@@ -1,74 +1,199 @@
+
+function P({ children }: PropsWithChildren): JSX.Element {
+  return (
+    <StyledText as="p" desktopStyle="bodyDefaultRegular" color={COLORS.black90}>
```
need `as="p"`?
i think if passing desktopStyle, we don't need to use `as`.

koji commented on protocol-designer/src/components/organisms/AutoAddPauseUntilTempStepModal/index.tsx:
```diff
@@ -1,74 +1,199 @@
+        })}
+        titleElement1={
+          <Icon name="alert-circle" size="1.25rem" color={COLORS.yellow50} />
+        }
```
i think you can use `type` instead of using titleElement.
since a 16px gap is applied to the titleElement, placing an icon next to the title would no longer align with the intended design.

SyntaxColoring replied:
Nice catch, TY. Changed to `type="warning"`.

I think `Modal` still puts a 16px gap around the `type="warning"` icon, though. Maybe that's a separate bug in `Modal`?

SyntaxColoring commented on protocol-designer/src/components/organisms/AutoAddPauseUntilTempStepModal/index.tsx:
```diff
@@ -1,74 +1,199 @@
+        case 'thermocyclerLid':
+          return 'modal:auto_add_pause_until_temp_step.thermocycler_lid:body'
+        case 'thermocyclerProfile':
+          return 'modal:auto_add_pause_until_temp_step.thermocycled_profile:body'
```
Typo, `thermocycled`?

SyntaxColoring replied:
`.body` instead of `:body`?

SyntaxColoring committed:
Rename type -> modalType.

SyntaxColoring committed:
Remove unnecessary `as="p"`.

SyntaxColoring committed:
Use modal `type` instead of `titleElement`.

SyntaxColoring committed:
Fix messed-up i18n keys.

- `thermocycler_profile.title` was missing entirely, oops!
- Use `.`, not `:`, as the key separator. `:` works too, apparently, but it's good to be consistent.
- Fix a "thermocycled" typo.

SyntaxColoring committed:
Lint.

SyntaxColoring committed:
Merge branch 'edge' into those_sweet_sweet_modal_dialogs

Resolve formatting conflicts in protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/index.tsx.

jerader commented on protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/index.tsx:
```diff
@@ -39,6 +40,7 @@ interface StateProps {
   isPristineSetHeaterShakerTempForm: boolean
   invariantContext: InvariantContext
   formData?: FormData | null
+  enableConcurrentModuleActions: boolean
```
nit: can this prop be above `formData` since `formData` is optional

jerader commented on protocol-designer/src/components/organisms/AutoAddPauseUntilTempStepModal/index.tsx:
```diff
@@ -1,74 +1,194 @@
+  }
+}
+
+function P({ children }: PropsWithChildren): JSX.Element {
```
nit, can you make this name bit more descriptive?

jerader reviewed (commented):
lgtm! left a few minor comments

ncdiehl11 reviewed (approved):
lgtm after addressing Jethary's comments

SyntaxColoring committed:
Rename <P> -> <BodyParagraph>.

SyntaxColoring committed:
Reorder optional formData prop to be last.
